@trycourier/courier-react-17/src/utils/render.tsx

```diff
@@ -0,0 +1,22 @@
+import { ReactNode } from "react";
+import { render } from "react-dom";
+
+/**
+ * Converts a React node to an HTMLElement.
+ * @param node - The React node to convert.
+ * @returns The converted HTMLElement.
+ */
+export function reactNodeToHTMLElement(node: ReactNode): HTMLElement {
+  const container = document.createElement('div');
+
+  render(node, container);
```

Comment:
`flushSync` doesn't exist in React 17, and rendering is synchronous already: https://react.dev/reference/react-dom/client/createRoot#root-render-caveats